### PR TITLE
Update validate script

### PR DESCRIPTION
**Pull request by forabi**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
     "sourceMap": true,
     
-    "allowJs": true,
+    "allowJs": false,
```
Comment (msafih) [inline tsconfig.json]: Why is this?

File: validateFileNames.js [hunk 2]
```diff
@@ -67,23 +75,28 @@ if (filesInViolation.length > 0) {
...
 
 function getFiles() {
   // Let's only validate files managed by git
-  const files = shelljs.exec('git ls-files', { silent: true });
-
-  return files.split('\n').filter(file => {
-    // remove empty strings from the array and remove files in ignored paths
-    return file.length !== 0 && !isIgnored(file);
-  });
+  const { code, stdout } = shelljs.exec('git ls-files', { silent: true });
```
Comment (msafih) [inline validateFileNames.js]: Hmmm, for me the return of `shelljs.exec('git ls-files', { silent: true })` was a string? 🤔

Comment (forabi) [inline validateFileNames.js]: This is strange. It always was an object for me (even in `deploy.js` of [hollowverse/hollowverse-docker](https://github.com/hollowverse/hollowverse-docker))

screenshot_20170820_232052

File: tsconfig.json (hunk 1 above)
Comment (forabi) [inline tsconfig.json]: I've described the problem in the commit message: "TypeScript seems to be confusing JS files with TS files and doing strict checks on things that cannot be typed in JS." At least, VS Code was. Once I turned off this, it stopped doing the strict checks and only showed a subset of the errors.

File: validateFileNames.js (hunk 2 above)
Comment (msafih) [inline validateFileNames.js]: Well, okay. But I'm positive that the earlier implementation was working.

Perhaps it's a monkey patched string that's created with `new String('whatever')`

`typeof new String('whatever')` is an `object`.

Comment (forabi) [inline validateFileNames.js]: It seems so

```
> shelljs.exec('git ls-tree', { silent: true })
{ [String: '']
  stdout: '',
  stderr: '...',
}     
``

**Pull request merged**